v1/factories.go

```diff
@@ -12,12 +12,16 @@ import (
 )
 
 // BrokerFactory creates a new object of brokers.Interface
-// Currently only AMQP broker is supported
+// Currently only AMQP/S broker is supported
 func BrokerFactory(cnf *config.Config) (brokers.Interface, error) {
 	if strings.HasPrefix(cnf.Broker, "amqp://") {
 		return brokers.NewAMQPBroker(cnf), nil
 	}
 
+	if strings.HasPrefix(cnf.ResultBackend, "amqps://") {
+		return backends.NewAMQPBackend(cnf), nil
```

Comment:
Also rebase against master as there have been quite a lot of changes in last couple of weeks. After months of low activity I finally managed to get some free time last two weeks. So lots of refactoring and changes in the code. Plus new features.